civis/ml/_model.py

```diff
@@ -910,8 +898,9 @@ def _create_custom_run(self, template_id, job_name=None, table_name=None,
                 # modify via arguments if users give a non-default value.
                 script_arguments[key] = value
         if self.git_token_name:
-            script_arguments['GITHUB'] = _get_credential(
-                self.git_token_name)
+            cred = find_one(self._client.credentials.list(),
+                            name=self.git_token_name)
```

Comment:
I'd add the `type='Custom'` parameter to the `.list()` call. That will reduce the number of credentials returned, and reduce the chance of name collisions. I think it's also worth verifying that the name is unique and giving an error if not. There could easily be more than one custom credential with the same name.